code/controller/controller.cpp

```diff
@@ -1,15 +1,38 @@
 #include "controller.h"
 
+Vector2i Controller::toVec2(Dir dir) {
+	Vector2i vector_dir(0, 0);
 
-Controller::Controller(Mode)
-{
-
-}
-
-Controller::~Controller()
-{
-
+	switch (dir) {
+	case UP:
+		vector_dir.y_point = 1;
+		break;
+	case DOWN:
+		vector_dir.y_point = -1;
+		break;
+	case RIGHT:
+		vector_dir.x_point = 1;
+		break;
+	case LEFT:
+		vector_dir.x_point = -1;
+		break;
+	default:
+		Logger::log("Invalid direction specified for movement: " + dir, Logger::ERROR);
+		return vector_dir;
+	}
+	return vector_dir;
 }
 
-void Controller::move(Dir) {}
-void Controller::change_mode(Mode) {}
+void Controller::invertAxis(Axis axis) {
+	switch (axis) {
+	case X:
+		m_invert_x *= -1;
+		break;
+	case Y:
+		m_invert_y *= -1;

```

Comment:
Hmm.. or actually I'm not sure what the best naming is. Think we should keep a note of this because it's in a decent amount of places in the code and commenting every instance isn't great?